pkg/monitortests/clusterversionoperator/requiredsccmonitortests/monitortest.go

```diff
@@ -0,0 +1,131 @@
+
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var defaultSCCs = sets.NewString(
+	"anyuid",
+	"hostaccess",
+	"hostmount-anyuid",
+	"hostnetwork",
+	"hostnetwork-v2",
+	"nonroot",
+	"nonroot-v2",
+	"privileged",
+	"restricted",
+	"restricted-v2",
+)
+
+type requiredSCCAnnotationChecker struct {
+	kubeClient kubernetes.Interface
+}
+
+func NewAnalyzer() monitortestframework.MonitorTest {
+	return &requiredSCCAnnotationChecker{}
+}
+
+func (w *requiredSCCAnnotationChecker) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	var err error
+	w.kubeClient, err = kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w *requiredSCCAnnotationChecker) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	if w.kubeClient == nil {
+		return nil, nil, nil
+	}
+
+	namespaces, err := w.kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	junits := []*junitapi.JUnitTestCase{}
+	for _, ns := range namespaces.Items {
+		if !strings.HasPrefix(ns.Name, "openshift") && !strings.HasPrefix(ns.Name, "kube-") && ns.Name != "default" {
+			continue
+		}
+
+		suggestedSCC := ""
+		switch ns.Labels["openshift.io/run-level"] {
+		case "0", "1":
+			suggestedSCC = "privileged"
+		}
+
+		pods, err := w.kubeClient.CoreV1().Pods(ns.Name).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		failures := make([]string, 0)
+		for _, pod := range pods.Items {
+			if _, exists := pod.Annotations[securityv1.RequiredSCCAnnotation]; exists {
+				continue
+			}
+
+			// suggest the assigned SCC only if it belongs to the default set of SCCs
+			if len(suggestedSCC) == 0 && defaultSCCs.Has(pod.Annotations[securityv1.ValidatedSCCAnnotation]) {
+				suggestedSCC = pod.Annotations[securityv1.ValidatedSCCAnnotation]
+			}
+
+			ownerRefs := make([]string, len(pod.OwnerReferences))
+			for i, or := range pod.OwnerReferences {
+				ownerRefs[i] = fmt.Sprintf("%s/%s", strings.ToLower(or.Kind), or.Name)
+			}
+
+			msg := fmt.Sprintf("annotation missing from pod '%s'; suggested required-scc: '%s'", pod.Name, suggestedSCC)
+			if len(ownerRefs) > 0 {
+				msg += fmt.Sprintf("; owners: %s", strings.Join(ownerRefs, ", "))
+			}
+			failures = append(failures, msg)
+		}
+
+		testName := fmt.Sprintf("[sig-auth] all workloads in ns/%s must set the '%s' annotation", ns.Name, securityv1.RequiredSCCAnnotation)
+		junit := &junitapi.JUnitTestCase{Name: testName}
+		junits = append(junits, junit)
```

Comment:
please add after you're done mutating. Makes it easier to read

Reply:
oh, wow.  You're trying to do success adn then failure.  Yeah, definitely refactor.  Add a different instance entirely.